### ISSUE TITLE
`padding-line-between-statements`: Allow more control regarding exports

### ISSUE THREAD

**Issue by Harpush** [eslint-stylistic/eslint-stylistic#1130]
### Clear and concise description of the problem

I wanted to add line breaks around blocks so I did:
```
{ blankLine: 'always', prev: '*', next: 'block-like' },
{ blankLine: 'always', prev: 'block-like', next: '*' }
```
But that caused line break between overload and implementation.. So I added:
```
{ blankLine: 'never', prev: 'function-overload', next: 'function' },
```
But didn't work... after trying some time to figure out why I understood it is related to exports.
I have yet to solve my need but I think anyway that exports should be handled differently.
My case:
```ts
export function abc() {

}

export function test(a: number, b: string): void;
export function test(a: number, b: string): void;
export function test(a: number | string, b: string | number): void {
  return;
}

export function bcd() {
  
}
```
I want to ALWAYS enforce line break between `abc` and `test`, between `test` and `bcd` but NEVER between `test` overloads and implementation.

### Suggested solution

Allow passing a boolean flag - `ignoreExports`. So the above example can be:
```
{ blankLine: 'always', prev: '*', next: 'block-like' , ignoreExports: true},
{ blankLine: 'always', prev: 'block-like', next: '*' , ignoreExports: true},
{ blankLine: 'never', prev: 'function-overload', next: 'function' , ignoreExports: true},
```
Essentially allowing to treat in every rule the export keyword as non existent.
Also can be more complex: `exportsHandling: 'ignore' | 'require'` giving even more control.

### Alternative

I guess there is a workaround but it's probably more complicated and for my use case I have yet to find one...

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue
